python/graphstorm/dataloading/metadata.py

```diff
@@ -0,0 +1,954 @@
+                'Some edge types in edge feature dimensions: '
+                f'{efeat_dims} are not in the edge type '
+                f'list: {etypes}.')
+        self._efeat_dims = efeat_dims
+
+    # getters
+    def is_homo(self):
+        """ Check if the graph metadata is for a homogeneous graph.
+
+        Return
+        -------
+        bool: if the graph metadata is for a homogeneous graph.
+        """
+        return self._gtype == HOMO_GRAPH_TYPE
+
+    def get_ntypes(self):
+        """ Get node types.
+
+        Return
+        -------
+        list: graph node types in a list of strings.
+        """
+        return self._ntypes
+
+    def has_ntype(self, ntype):
+        """ Check if a node type exists in the graph metadata.
+
+        Parameter
+        ---------
+        ntype: str
+            The node type name.
+
+        Return
+        -------
+        bool: if the graph metadata contains the given node type.
+        """
+        return ntype in self._ntypes
+
+    def get_etypes(self):
+        """ Get edge types.
+
+        Return
+        -------
+        list: graph node types in a list of 3-element tuples.
+        """
+        return self._etypes
+
+    def has_etype(self, etype):
+        """ Check if an edge type exists in the graph metadata.
+
+        Parameter
+        ---------
+        etype: str
+            The edge type name in a three element tuple.
+
+        Return
+        -------
+        bool: if the graph metadata contains the given edge type.
+        """
+        return etype in self._etypes
+
+    def get_nfeat_all_dims(self, ntype):
+        """ Get all feature dimensions of the given node type.
+
+        Parameter
+        ---------
+        ntype: str
+            The node type name.
+
+        Return
+        -------
+        dict or None: the feature dimensions of the given ntype in the format of a dictionary, or
+                      None if either the graph metadata has no node feature or the given ntype has
+                      no feature.
+        """
+        if self._nfeat_dims is not None:
+            nfeat_all_dims = self._nfeat_dims.get(ntype, None)
+        else:
+            nfeat_all_dims = None
+
+        return nfeat_all_dims
+
+    def get_efeat_all_dims(self, etype):
+        """
+        Parameter
+        ---------
+        etype: str
+            The edge type name in a three element tuple.
+
+        Return
+        ------
+        list or None: the feature dimensions of the given etype in the format of a dictionary, or
+                      None if either the graph metadata has no edge feature or the given etype has
+                      no feature.
+        """
+        if self._efeat_dims is not None:
+            efeat_all_dims = self._efeat_dims.get(etype, None)
+        else:
+            efeat_all_dims = None
+
+        return efeat_all_dims
+
+    # output functions
+    def to_dict(self):
```

Comment:
It seems like this serializes the class which means we need a well-defined, versioned schema to ensure we maintain backwards compatibility as new entries are added to the representation.

Reply:
Added a version  number of this JSON to keep track its evolution.